fast_llm/data/dataset/gpt/sampled.py

```diff
@@ -362,6 +362,183 @@ def _load_yaml_data(self, data: dict[str, typing.Any]):
         self._unshuffled_documents = data["unshuffled_epochs"] * self._documents_per_epoch
 
 
+class GPTSampledIndexedPaddedDataset(SampledDataset):
```

Comment:
There is a lot of code in common with `GPTSampledIndexedDataset`. Could we merge the two classes, or at least create a common base class to reduce duplication?